### DISCUSSION TITLE
preview buffer jittery behaviour

### DISCUSSION THREAD

**Discussion by ville-h** [minad/consult#985] [Q&A]
When using constult and consult-eglot `consult-eglot-symbols` the preview buffers behave in an undesirable way. Every key press the buffer for the first symbol in the list is briefly flashed before showing the intended preview buffer. It is rather jarring to type and watch the buffers flash.

To further explain If you invoke `consult-eglot-symbols` a preview buffer B0 is shown for the first symbol "abc" of the candidate symbol list. You press 'd' and land on preview buffer B1 on symbol "def". Press 'e' and the preview buffer B0 is briefly flashed before landing back on preview buffer B1 on symbol "def" again. Press 'f' and the preview buffer for B0 is briefly flashed before landing back on the preview buffer B1 on symbol "def" again.

Is there some way to disable this behaviour that keeps flashing the preview buffer B0 unnecessarily, but does not disable the preview feature?

**Comment by minad** [comment]
I haven't tried consult-eglot yet. Maybe the problem goes away if you use a short delay for the preview? cc @mohkale

**Reply by mohkale** [reply]
I haven't actually noticed any delay so I'd need @ville-h to confirm if any changes fix the problem. Do you have any examples of this @minad ? I've been trying to stick relatively close to consult--grep-state so would this be scheduling a callback after a timeout in there?

For reference this is the current state function: https://github.com/mohkale/consult-eglot/blob/64262e72452f8fe6dd49d31bcdd4bd577b7d682d/consult-eglot.el#L179

**Reply by minad** [reply]
@mohkale I didn't mean a code change in consult-eglot, but rather using a small value for preview `:debounce`, e.g., 0.2s, see https://github.com/minad/consult?tab=readme-ov-file#live-previews.

**Reply by mohkale** [reply]
Ah, I see, @ville-h can you try customizing like this and see if it fixes your issue? :thinking:.

**Reply by ville-h** [reply]
I am not a lisp programmer so it is unclear what is the _exact_ code snippet I am supposed to put into my init.el? Something I can directly copy-and-paste. I am using use-package if that matters.

**Reply by mohkale** [reply]
@ville-h 

Try.

```elisp
(with-eval-after-load 'consult-eglot
  (consult-customize consult-eglot-symbols :preview-key '(:debounce 0.4 any)))

```

**Comment by ville-h** [comment]
I've now tried mohkale's suggestion which appears to work. I played around with the values and in this particular instance values >=0.2 seemed to work around the issue. Even at 0.2 the usability feels laggy.

If I understand the fix right it sort of papers over the issue rather than properly fixes it. If some project was particularly complex and returning/resolving the symbol information takes long then the debounce value might have to be increased?

Also this still leaves a similar issue with the candidate list display which I hadn't noticed before. The candidate list also displays some kind of unrelated "default" state when typing/erasing characters before displaying the refined candidate list.
